packages/terra-dev-site/src/webpack/plugin/SitePlugin.js

```diff
@@ -341,6 +359,72 @@ class SitePlugin {
       oneTimeSetupComplete = false;
       siteRegistry = {};
     });
+
+    // Evaluate all URLS found in the site for correctness.
+    compiler.hooks.done.tap('terra-dev-site-plugin', () => {
+      // Convert generated links for all sites into a single list.
+      const validRelativeUrls = Object.values(generatedBootstrapData).reduce((acc, urls) => acc.concat(urls), []);
+      // add the favicon since it's valid too.
+      validRelativeUrls.push(`${publicPath.slice(0, -1)}/favicon.ico`);
+
+      // Log formatting
+      const invalidUrlLog = (url) => `${chalk.blue(url.url)} in:\n ${chalk.yellow(url.file.history[0].replace(url.file.cwd, ''))} at line ${url.node.position.start.line}`;
+
+      // Create a promise array to allow all logs to happen at the same time.
+      const resultPromises = [];
+      const queueLength = urlQueue.length;
+      for (let index = 0; index < queueLength; index += 1) {
+        const url = urlQueue.shift();
+
+        if (url.url.startsWith('/')) {
+          // for relative urls check against the know list of valid relative urls.
+          // Remove query params or hash links from the end of the url.
+          const urlWithoutHash = url.url.replace(/[#?].*$/, '');
+          if (!validRelativeUrls.includes(urlWithoutHash)) {
+            resultPromises.push(Promise.resolve(`${chalk.red('[Broken Link]')} ${invalidUrlLog(url)}`));
+          }
+        } else if (url.url.startsWith('https')) {
+          // required to not assume https urls are invalid.
+          // We cannot validate engineering.cerner.com urls without additional work because they use the 404 status to redirect.
+          // resultPromises.push(new Promise((resolve) => {
+          //   https.get(url.url, (response) => {
+          //     if (response.statusCode >= 400) {
+          //       resolve( `${chalk.red('[Broken Link]')} ${chalk.blue(url.url)} in:\n ${chalk.yellow(url.file.history[0].replace(url.file.cwd, ''))} at line ${url.node.position.start.line}`);
+          //     } else {
+          //       resolve();
+          //     }
+          //   });
+          // }));
+        } else if (url.url.startsWith('http')) {
+          // required to not assume http urls are invalid.
+          // resultPromises.push(new Promise((resolve) => {
+          //   http.get(url.url, (response) => {
```

Comment:
it's really annoying that https can't handle http calls or that they're separate at all... thanks node.

Reply:
Or we could use a light-weight third-party lib but it's nice to just use the native since it's already available.